rules-java-ee/addon/src/main/java/org/jboss/windup/rules/apps/javaee/TechnologyUsageStatisticsService.java

```diff
@@ -0,0 +1,384 @@
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVICES_HIBERNATE_MAPPINGFILES, countByType(HibernateMappingFileModel.class));
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVICES_HIBERNATE_SESSIONFACTORIES,
+        // countByType(HibernateSessionFactoryModel.class));
+        //
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVERRESOURCES_DB_JDBCDATASOURCES, this.countByType(DataSourceModel.class, new
+        // HashMap<String, Serializable>(){{
+        // put(DataSourceModel.IS_XA, false);
+        // }}));
+        //
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVERRESOURCES_DB_XAJDBCDATASOURCES, this.countByType(DataSourceModel.class, new
+        // HashMap<String, Serializable>(){{
+        // put(DataSourceModel.IS_XA, true);
+        // }}));
+        //
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVICES_HTTP_JAX_RS, this.countByType(JaxRSWebServiceModel.class));
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVICES_HTTP_JAX_WS, this.countByType(JaxWSWebServiceModel.class));
+        //
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVERRESOURCES_MSG_JMS_QUEUES, this.countByType(JmsDestinationModel.class,
+        // JmsDestinationModel.DESTINATION_TYPE, JmsDestinationType.QUEUE.name()));
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVERRESOURCES_MSG_JMS_TOPICS, this.countByType(JmsDestinationModel.class,
+        // JmsDestinationModel.DESTINATION_TYPE, JmsDestinationType.TOPIC.name()));
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVERRESOURCES_MSG_JMS_CONNECTIONFACTORIES,
+        // this.countByType(JmsConnectionFactoryModel.class));
+        //
+        // //stats.setStatsServerResourcesSecurityRealms(item(countByType(.class)));
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_SERVERRESOURCES_JNDI_TOTALENTRIES, this.countByType(JNDIResourceModel.class));
+        //
+        // // Not sure how to get this number. Maybe JavaClassFileModel.getJavaClass() ?
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_JAVA_CLASSES_ORIGINAL, this.countJavaClassesOriginal());
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_JAVA_CLASSES_TOTAL, this.countByType(JavaClassModel.class));
+        //
+        // // We are not able to tell which of the jars are original. We can substract known opensource libs.
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_JAVA_JARS_ORIGINAL, this.diff(
+        // this.countByType(JarArchiveModel.class),
+        // this.countByType(IdentifiedArchiveModel.class))
+        // );
+        // technologyUsage.put(TechnologyUsageStatisticsModel.STATS_JAVA_JARS_TOTAL, this.countByType(JarArchiveModel.class));
+
+        return this.groupByProjectModel(technologyUsage);
```

Comment:
How would this method work? Is it obsolete now, or to be fixed later?

Reply:
Yes, this is obsolete. I left the commented out code in place in case it is useful for when the rules are created. Ideally, the rules should be in xml, IMO, rather than something that requires code changes.